.github/workflows/node.js.yml

```diff
@@ -0,0 +1,109 @@
+      # - master
+  pull_request:
+    branches:
+      - jp-gha-conversion
+      - staging
+      # - master
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Testing the app build + Prettier/Lint checks
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+
+      - name: Node.js setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: 💅 Prettier check
+        run: yarn prettier:check
+
+      - name: 💨 Lint check
+        run: yarn lint
+
+      - name: 🏗️ Build app
+        run: yarn build --prod
+
+  deploy:
+    name: Deploy
+    if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/jp-gha-conversion')}}
+    runs-on: ubuntu-latest
+    environment:
+      name: ${{ (github.ref == 'refs/heads/master' && 'prod') || 'stage' }}
+    needs: build
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+
+      - name: Install AWSCLI
+        run: pip install --user awscli
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_ARN }}
+
+      - name: Node.js setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: 🏗️ Build app
+        run: yarn build --prod
+
+      - name: Copy embed.js
+        run: cp embed/embed.js dist/embed.js
+
+      - name: Copy example.html
+        run: cp embed/example.html dist/example.html
+
+      - name: Copy mobile directory
+        run: cp -r mobile/ dist/mobile/
+
+      - name: Update embed host (staging)
+        if: github.ref == 'refs/heads/staging'
+        run: sed -i "s%$appDomainPlaceholder%$appDomain%g" dist/embed.js
+        env:
+          appDomainPlaceholder: '{appDomain}'
+          appDomain: 'https://stage.knowgod.com'
+
+      - name: Deploy to s3 (staging)
+        if: github.ref == 'refs/heads/staging'
+        run: |
+          aws s3 sync dist/knowgod s3://$S3_BUCKET --acl 'public-read' --exclude '*.html' --cache-control 'public, max-age=31536000' --delete
+          aws s3 sync dist/knowgod s3://$S3_BUCKET --acl 'public-read' --include '*.html' --cache-control 'no-cache' --delete
+          aws s3 cp dist/embed.js s3://$S3_BUCKET --acl 'public-read' --cache-control 'no-cache'
+          aws s3 cp dist/example.html s3://$S3_BUCKET --acl 'public-read' --cache-control 'no-cache'
+          aws s3 sync dist/mobile s3://$S3_BUCKET --acl 'public-read' --cache-control 'no-cache' --content-type 'application/json'
+        env:
+          S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
```

Comment:
so a better way to do this instead of having to put `s3://` on every line is to just include the `s3://` when you are defining the `S3_BUCKET` environment variable

Reply:
Should I go ahead and update that in Terraform or not worry about it for now?

Reply:
don't do that in terraform, do that in this workflow.

For reference you can customize the generation of environment variables like this: https://github.com/CruGlobal/android-gto-support/blob/master/.github/workflows/build.yml#L220